src/DGSM.jl

```diff
@@ -0,0 +1,50 @@
+
+mutable struct dgsm
+    a::Float64
+    absa::Float64
+    asq::Float64
+end
+
+"""
+The inputs to the function DGSM are as follows:
```

Comment:
Aren't there other DGSMs? What's the name this one usually goes by?

Reply:
There are other DGSM's. But all DGSM's are essentially integrals and I couldn't find specific names for it. The most important DGSM is E(a^2) where a is partial derivative of f wrt x_i and it was just named as v_i = E(a^2). I searched many papers but I couldn't find the specific names of various DGSM's

Reply:
So I am naming this function as DGSM_Vi.